crates/cursor-core/src/services/flagged_chunk.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone)]
+pub struct FlaggedChunk {
+    pub data: Vec<u8>,
+    pub flag: u8,
+}
+
+impl FlaggedChunk {
```

Comment:
Maybe we should call it `EnvelopedMessage` or something? I found it from here:  https://connect.build/docs/protocol.